### ISSUE TITLE
the loss is nan

### ISSUE THREAD

**Issue by ernestzhang** [jumon/whisper-finetuning#13]
After more than 1400 steps, the loss becomes nan . params are follow:
{
    "train_json": "train.json",
    "dev_json": "dev.json",
    "batch_size": 4,
    "dev_batch_size": 16,
    "no_timestamps_training": false,
    "prompt_use_rate": 0.5,
    "no_timestamps_rate": 0.5,
    "save_dir": "output",
    "device": "cuda",
    "model": "large-v2",
    "train_only_decoder": true,
    "lr": 0.0001,
    "accum_grad_steps": 64,
    "warmup_steps": 500,
    "max_grad_norm": 1.0,
    "train_steps": 10000,
    "eval_steps": 200,
    "save_all_checkpoints": false,
    "seed": 42,
    "use_adam_8bit": true
}
<img width="868" alt="1" src="https://github.com/jumon/whisper-finetuning/assets/1472398/b54e1664-563d-4ba0-9ef8-1247453efa5c">

**Comment by jumon** [comment]
How did your loss look prior to step 1400? Was there a decrease in loss or no improvement?

**Comment by ernestzhang** [comment]
Before that, the loss was not very stable, sometimes it became larger, sometimes it was smaller, but it was generally convergent. I tried again today, and it became nan when it was 200+ steps. I just changed a little code to adapt to multi-graphics card training, I think it should not matter. Today I restored the code to its original state and tried again. Here is where my code changed:
<img width="1171" alt="2" src="https://github.com/jumon/whisper-finetuning/assets/1472398/1553c71e-2a38-4fcf-9a3f-1d8904464938">

**Comment by ernestzhang** [comment]
All the same with the original code
<img width="564" alt="3" src="https://github.com/jumon/whisper-finetuning/assets/1472398/8876b277-d2f4-4e8e-876d-fd38afa6c470">

**Comment by jumon** [comment]
Thank you for the details! The instability of the loss could potentially be attributed to insufficiently optimized hyper-parameters. In scenarios like this, I would start by tuning the tiny model, as larger models tend to be more challenging to stabilize and fine-tune. Nevertheless, it is also possible that there are problems within my code. It's important to note that this project is still a work in progress.
